### PR TITLE
host_wasmtime: support multiple sensor devices per backend

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: guest_c/main.c [hunk 1]
```diff
@@ -211,7 +211,7 @@ exports_wasi_sensor_interface_main()
 
         fprintf(stderr, "C guest started\n");
 
-        const char *sensor = "dummy";
+        const char *sensor = "dummy:dummy";
```
Comment (ayakoakasaka) [inline guest_c/main.c:214]: You mean we can specify
dummy:image
dummy:depth
dummy:imu

if we create dummy image device, dummy depth device, dummy imu device ?

File: host_wasmtime/src/main.rs [hunk 2]
```diff
@@ -193,19 +198,16 @@ impl<T: WasiSensorView> wasi::sensor::sensor::HostDevice for T {
...
-            }),
-            #[cfg(feature = "nokhwa")]
-            "nokhwa" => Box::new(match NokhwaDevice::new() {
-                Err(e) => return Ok(Err(e)),
-                Ok(i) => i,
-            }),
-            _ => return Ok(Err(wasi::sensor::sensor::DeviceError::NotFound)),
+        let v: Vec<&str> = device_name.split(":").collect();
+        if v.len() != 2 {
+            return Ok(Err(wasi::sensor::sensor::DeviceError::NotFound));
```
Comment (ayakoakasaka) [inline host_wasmtime/src/main.rs:203]: "aaa:bbb"  would be mandatory ?

File: guest_c/main.c (hunk 1 above)
Comment (ayakoakasaka) [reply guest_c/main.c:214]: Do you think it makes sense ?

`nokhwa:/dev/video0 `

Comment (yamt) [reply guest_c/main.c:214]: > You mean we can specify dummy:image dummy:depth dummy:imu

the dummy device ignores the latter part for now.

> if we create dummy image device, dummy depth device, dummy imu device ?

it's possible.

Comment (yamt) [reply guest_c/main.c:214]: > Do you think it makes sense ?
> 
> `nokhwa:/dev/video0 `

sort of. except that /dev/video0 seems like a very linux specific identifier.

File: host_wasmtime/src/main.rs (hunk 2 above)
Comment (yamt) [reply host_wasmtime/src/main.rs:203]: in this particular implementation, yes.
in the wasi-sensor api, no.

File: guest_c/main.c (hunk 1 above)
Comment (ayakoakasaka) [reply guest_c/main.c:214]: understood. I agree as the spec.

**Review by ayakoakasaka — APPROVED** [review]

**Pull request merged**